src/python_framework_v02/troute/nhd_io.py

```diff
@@ -1084,3 +1086,73 @@ def build_coastal_ncdf_dataframe(coastal_ncdf):
     with xr.open_dataset(coastal_ncdf) as ds:
         coastal_ncdf_df = ds[["elev", "depth"]]
         return coastal_ncdf_df.to_dataframe()
+
+def lastobs_df_output(dt,
+    nts,
```

Comment:
can we pass the run_parameters dictionary instead of these individual variables? The run_parameters dict contains t0, which is critical for timestamping the lastobs output. Maybe the run_parameters dict is inconsistently structured between v2 and v3? What do you think?

Reply:
for timestamping i just sent `t0 + timedelta(seconds = dt * nts)` for the timestamp input itself. I do believe v2 and v3 have some differences in the structuring of the run_parameters. It may just be easier to leave it since they are all available in main? Up to you.